netbox/extras/jobs.py

```diff
@@ -100,5 +103,19 @@ def run(self, data, request=None, commit=True, **kwargs):
 
         # Execute the script. If commit is True, wrap it with the event_tracking context manager to ensure we process
         # change logging, event rules, etc.
-        with event_tracking(request) if commit else nullcontext():
-            self.run_script(script, request, data, commit)
+        branch = None
+        if settings.BRANCHING_BACKEND:
+            try:
+                branching_cls = import_string(settings.BRANCHING_BACKEND)
+            except AttributeError:
+                logger = logging.getLogger(f"netbox.scripts.{script.full_name}")
+                message = _("Failed to import configured BRANCHING_BACKEND: ") + settings.BRANCHING_BACKEND
+                logger.error(message)
+                raise ImproperlyConfigured(message)
+
+            branching_backend = branching_cls()
+            branch = branching_backend.get_active_branch(request)
+
+        with branching_backend.activate_branch(branch) if branch else nullcontext():
+            with event_tracking(request) if commit else nullcontext():
```

Comment:
Instead of checking for a branching context manager explicitly, it would be more extensible to allow the configuration of additional arbitrary context managers to be [chained](https://stackoverflow.com/a/30981865/1516387) with `event_tracking()`. This could provide future functionality unrelated to branching.